torch/distributed/optim/named_optimizer.py

```diff
@@ -268,6 +273,16 @@ def add_param_group(self, param_group: Any) -> None:
             "add_param_group not supported yet and might be implemented soon."
         )
 
+    def _pre_load_state_dict(self, state_dict) -> None:
+        if isinstance(self.module, FSDP):
+            return FSDP._load_optim_state_dict_pre_hook(self.module, self._optimizer, state_dict)
```

Comment:
why are we returning if it is FSDP, shouldn't we do the preprocessing?

Reply:
`_load_optim_state_dict_pre_hook` does the preprocessing here.